drivers/sensor/st/lsm6dsv16x/lsm6dsv16x.c

```diff
@@ -501,6 +725,12 @@ static int lsm6dsv16x_accel_get_config(const struct device *dev,
 	struct lsm6dsv16x_data *data = dev->data;
 
 	switch (attr) {
+#ifdef CONFIG_LSM6DSV16X_SELF_TEST
+	case SENSOR_ATTR_GET_SELF_TEST_RESULT:
+		val->val1 = (data->xl_st_result == 0) ? 0 : 1;
```

Comment:
So `lsm6dsv16x_gyro_self_test` and `lsm6dsv16x_accel_self_test` will return `-1` for failure and `0` for success, right? So then we convert it to `0` -> success, `1` -> failure? That seems a bit confusing, TBH. Also maybe we can define "pass" / "fail" constants in `include/zephyr/drivers/sensor/lsm6dsvxxx.h` to clarify the API?